pandas/tests/indexes/test_base.py

```diff
@@ -1131,6 +1131,18 @@ def test_get_indexer_strings(self):
         with pytest.raises(TypeError):
             idx.get_indexer(['a', 'b', 'c', 'd'], method='pad', tolerance=2)
 
+    def test_get_indexer_boolean_index_boolean_target(self):
+        boolean_idx = pd.Index([True, False])
+        actual = boolean_idx.get_indexer([True, True, False])
+        expected = np.array([0, 0, 1])
+        tm.assert_numpy_array_equal(actual, expected, check_dtype=False)
```

Comment:
Why don't we check `dtype` here (we generally prefer to)?  I think if you set the `dtype` of `expected` to be `np.intp`, you should be okay.  Same for your test below.